client/app/scripts/superdesk-search/search.js

```diff
@@ -1020,6 +1026,75 @@
                             }, 0, false);
                         });
                     });
+
+                    /*
+                     * Converting to object and adding pre-selected subject codes to list in left sidebar
+                     */
+                    metadata
+                        .fetchSubjectcodes()
+                        .then(function () {
+                            scope.subjectcodes = metadata.values.subjectcodes;
+                            return tags.initSelectedFacets();
+                        })
+                        .then(function (currentTags) {
+                            scope.subjectitems = {
+                                subject: getSubjectCodes(currentTags)
+                            };
+                        });
+
+                    /*
+                     * Function for finding object by string array for subject codes
+                     */
+                    function getSubjectCodes(currentTags) {
+                        var queryArray = currentTags.selectedParameters, filteredArray = [];
+                        if (!$location.search().q) {
+                            return filteredArray;
+                        }
+                        for (var i = 0, queryArrayLength = queryArray.length; i < queryArrayLength; i++) {
+                            var queryArrayElement = queryArray[i];
+                            if (queryArrayElement.indexOf('subject.name') !== -1) {
+                                var elementName = queryArrayElement.substring(
```

Comment:
This looks like duplicated logic, we should be able to extract it into the service class.